### PR TITLE
{devel}[GCC/10.2.0,iccifort/2020.4.304] Boost v1.74.0

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

requires ~~https://github.com/easybuilders/easybuild-framework/pull/2193~~

### CODE REVIEW

**Comment by Micket** [comment]
Where do we stand on the serial + parallel Boost issue?

I think the Boost case is quite simple (as it's just additional libraries, so, no potential ABI issues. It is also be far the most useful thing to have a serial version off as well, since it's a hugely common dependency for non-MPI applications  (much more so than netCDF and HDF5).

**Comment by boegel** [comment]
@Micket We should check with @lexming on that, see https://github.com/easybuilders/easybuild-easyblocks/pull/2167 .

It may be too early for the `2020b`, but maybe it's just a small additional effort to be able to make the split.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2226
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3117.skitty.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/8bb112279a1ef396a756df7f5d5bac51 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
Using easyblocks from PR(s) https://github.com/easybuilders/easybuild-easyblocks/pull/2226
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node2685.swalot.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/21503bd2cdd0d82107544f012d7cfe42 for a full test report.

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Comment by boegel** [comment]
@boegelbot please test @ generoso
EB_ARGS="--include-easyblocks-from-pr 2226"

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
easybuild1.novalocal - Linux centos linux 8.2.2004, POWER, IBM pSeries (emulated by qemu) (power8le), Python 3.6.8
See https://gist.github.com/a458c014d4515b31320b4f72df90a6f5 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node3403.kirlia.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/9a8a2c696f9961d2d208cc6620ebf093 for a full test report.

File: easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+sources = ['%%(namelower)s_%s.tar.gz' % '_'.join(version.split('.'))]
+checksums = ['afff36d392885120bcac079148c177d1f6f7730ec3d47233aa51b0afa4db94a5']
+
+dependencies = [
+    ('bzip2', '1.0.8'),
+    ('zlib', '1.2.11'),
+    ('XZ', '5.2.5'),
+]
+
+configopts = '--without-libraries=python'
```
Comment (Micket) [inline easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb:20]:
```suggestion
configopts = '--without-libraries=python,mpi'
```

File: easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name = 'Boost'
+version = '1.74.0'
+
+homepage = 'https://www.boost.org/'
+description = """Boost provides free peer-reviewed portable C++ source libraries."""
+
+toolchain = {'name': 'gompi', 'version': '2020b'}
```
Comment (Micket) [inline easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb:7]: -> GCC 10.2.0

File: easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+dependencies = [
+    ('bzip2', '1.0.8'),
+    ('zlib', '1.2.11'),
+    ('XZ', '5.2.5'),
+]
+
+configopts = '--without-libraries=python'
+
+# also build boost_mpi
+boost_mpi = True
```
Comment (Micket) [inline easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb:23]: -

**Review by Micket — CHANGES_REQUESTED** [review]
and the same for the intel flavour

File: easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb (hunk 1 above)
Comment (Micket) [reply easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb:20]: or rather, we should probably do `boost_mpi = False` ?

File: easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb (hunk 3 above)
Comment (boegel) [reply easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb:23]: `*`?

Comment (Micket) [reply easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb:23]: That was a - sign when i wrote it. but, perhaps it should just be set to False. The easyblock kind of gives us to many ways to achieve the same thing i think.

File: easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb (hunk 1 above)
Comment (lexming) [reply easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb:20]: `boost_mpi = False` should indeed be preferred and my plan with https://github.com/easybuilders/easybuild-easyblocks/pull/2167 is that `--without-libraries=python` should not be needed if Python is already not present as a dependency.

File: easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb (hunk 3 above)
Comment (lexming) [reply easybuild/easyconfigs/b/Boost/Boost-1.74.0-gompi-2020b.eb:23]: I agree, the functionality of `boost_mpi` can easily be moved to the toolchainopt `usempi`, but that means updating all easyconfigs as well

**Comment by boegel** [comment]
@boegelbot please test @ generoso

**Review by Micket — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
node2301.phanpy.os - Linux centos linux 7.8.2003, x86_64, Intel(R) Xeon(R) CPU E5-2680 v3 @ 2.50GHz (haswell), Python 2.7.5
See https://gist.github.com/2b5b6f9f7d5d80d831f5bdce8e94b0f2 for a full test report.

**Comment by boegel** [comment]
@lexming @Micket Good to go?

**Comment by Micket** [comment]
Going in, thanks @boegel!

**Pull request merged**